llmfoundry/data/finetuning/tasks.py

```diff
@@ -325,8 +325,8 @@ def get_preprocessing_fn_from_str(
         return preprocessing_fn
 
     def build_from_hf(
-        self, cfg: DictConfig, max_seq_len: int,
-        tokenizer: PreTrainedTokenizerBase
+        self, dataset_name_or_path: str, split: str, max_seq_len: int,
+        tokenizer: PreTrainedTokenizerBase, proto_preprocessing_fn: Union[dict, DictConfig, str], hf_kwargs: dict[str, Any]
```

Comment:
@dakinggg I want to do some refactors to make args more explicit, rather than passing around cfgs, but I'm guess this is an api breaking change? Any suggestions on how to proceed? 😬

Reply:
I am ok with that type of breaking...its so important. And it doesn't break usage at the yaml level, which is the more important API to keep intact.